app/src/main/java/com/github/sdp/ratemyepfl/fragment/navigation/ReviewFragment.kt

```diff
@@ -1,93 +1,47 @@
-
-        setupTabNavigation()
-
-    }
-
-    private fun setupTabNavigation() {
-        // Initially set the corresponding tab
-        childFragmentManager.commit {
-            val fragment = fromTabToFragment(tabLayout.selectedTabPosition)
-            if (fragment != null) {
-                add(
-                    R.id.reviewTabFragment,
-                    fragment
-                )
-                setReorderingAllowed(true)
-                addToBackStack("tab")
-            }
-        }
-
-        tabLayout.addOnTabSelectedListener(tabManager)
-    }
-
-    private val tabManager =
-        onTabSelectedListener(
-            { tab ->
-                when (tab?.position) {
-                    0 -> changeTab<CourseTabFragment>()
-                    1 -> changeTab<ClassroomTabFragment>()
-                    2 -> changeTab<RestaurantTabFragment>()
-                }
-            },
-            {},
-            {})
-
-    private inline fun <reified T : Fragment> changeTab() {
-        childFragmentManager.commit {
-            replace<T>(R.id.reviewTabFragment)
-            setReorderingAllowed(true)
-            addToBackStack("tab")
-        }
+        viewPager = view.findViewById(R.id.reviewTabViewPager)
+        val fragments = ReviewableTabFragment.TAB.values()
+            .map { it.toFragment() }
+            .toList()
+
+        viewPager.offscreenPageLimit = fragments.size
+        val viewPagerAdapter = FragmentViewPagerAdapter(fragments, this)
+        viewPager.adapter = viewPagerAdapter
+
+        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
+            tab.text = ReviewableTabFragment.TAB
+                .values()[position]
+                .name
+        }.attach()
     }
 
-
-    companion object {
-        private fun fromTabToFragment(tabPosition: Int): Fragment? =
-            when (tabPosition) {
-                0 -> CourseTabFragment()
-                1 -> ClassroomTabFragment()
-                else -> null
-            }
-
-        private fun onTabSelectedListener(
-            tabSelectedHandler: TabHandler,
-            tabReselectedHandler: TabHandler,
-            tabUnselectedHandler: TabHandler
-        ): TabLayout.OnTabSelectedListener =
-            object : TabLayout.OnTabSelectedListener {
-
-                override fun onTabSelected(tab: TabLayout.Tab?) {
-                    tabSelectedHandler(tab)
-                }
-
-                override fun onTabReselected(tab: TabLayout.Tab?) {
-                    tabReselectedHandler(tab)
-                }
-
-                override fun onTabUnselected(tab: TabLayout.Tab?) {
-                    tabUnselectedHandler(tab)
-                }
-            }
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return super.onCreateView(inflater, container, savedInstanceState)
```

Comment:
Do you plan to add code in the future to this method? Otherwise it is a redundant overriding method.

Reply:
I'll remove it.